src/main/java/com/android/volley/AsyncCache.java

```diff
@@ -0,0 +1,42 @@
+package com.android.volley;
+
+import androidx.annotation.Nullable;
+
+/**
+ * AsyncCache is an abstract class which implements the Cache interface for backwards compatibility,
+ * although it is not meant to be used as a Cache directly. For each method in the Cache interface
+ * (get, put, initialize, invalidate, remove, clear), it defines a corresponding abstract version
+ * which takes a Callback to be invoked upon completion instead of returning the result directly.
+ * The methods in the cache interface are implemented by making calls to the asynchronous method and
+ * blocking until they complete.
+ */
+public abstract class AsyncCache {
+
+    public interface OnGetCompleteCallback {
+        void onGetComplete(@Nullable Cache.Entry entry);
```

Comment:
nit: Could you add a Javadoc for this method? It doesn't have to be super-detailed, but one thing I think would be good to include is when/why entry would be null, and what we do in the event of an error. Something like:

```java
/**
 * Invoked when the read from the cache is complete.
 *
 * @param entry The entry read from the cache, or null if the cache did not contain an entry for the given key or
 *         if the entry couldn't be read.
 */
```